craft_parts/filesystem_mounts.py

```diff
@@ -107,6 +118,36 @@ def first_maps_to_slash(
             raise ValueError("The first entry in a filesystem must map the '/' mount.")
         return value
 
+    @field_validator("root", mode="after")
+    @classmethod
+    def validate_entries_order(
```

Comment:
I think the code itself is fine but is there a reason to do this in craft-parts, as opposed to in Imagecraft itself? If I remember the flow correctly, if we do the validation only in craft-parts it means that it will only be done when the LifecycleManager is created, right? 

Or is this model also used in Imagecraft directly?